src/Streamly/Internal/Data/Fold.hs

```diff
@@ -556,17 +556,17 @@ stdDev = sqrt variance
 --
 -- @since 0.7.0
 {-# INLINABLE rollingHashWithSalt #-}
-rollingHashWithSalt :: (Monad m, Enum a) => Int -> Fold m a Int
+rollingHashWithSalt :: (Monad m, Enum a) => Integer -> Fold m a Integer
 rollingHashWithSalt salt = Fold step initial extract
     where
-    k = 2891336453
+    k = 2891336453 :: Integer
```

Comment:
This is outside of 32bit Int, so I converted all to Integer. Not sure that has any impact.

Reply:
Integer would make it too slow. We can use `Int64` instead.